### PR TITLE
Added hibernate-orm property: quarkus.hibernate-orm.validation.enabled

### LINKED ISSUE #30448: Ability to set validation mode using a hibernate-orm property
### Description

I am using quarkus platform version: 2.9.2.Final

I have a use-case where I need to disable the hibernate validation of entities but still be able to control it on demand, in a later step, by using an injected Validator.

It seems like this is currently possible only using a persistence.xml file where the property <property name="javax.persistence.validation.mode" value="NONE"/> is set.

Using a persistence.xml is a step back and adds some configuration burdens on the project this is why I'd prefer to have a property using which the validation mode could be controlled.

P.S. I've tried with: quarkus.hibernate-orm.unsupported-properties."javax.persistence.validation.mode"=NONE
but it still did not work

### Implementation ideas

Would be nice to have a hibernate-orm property: **quarkus.hibernate-orm.validation.mode**  that controls how the hibernate validator behaves.

### PR DESCRIPTION
**Pull request by mfpc**
- Fixes #30448

### CODE REVIEW

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/validation/JPAValidationDisabledTestCase.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.orm.MyEntity;
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+
+public class JPAValidationDisabledTestCase {
+
+    @Inject
+    EntityManager em;
```
Comment (gastaldi) [inline extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/validation/JPAValidationDisabledTestCase.java:19]: You can remove this

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java [hunk 2]
```diff
@@ -496,4 +503,17 @@ public enum IdentifierQuotingStrategy {
         ALL_EXCEPT_COLUMN_DEFINITIONS,
         ONLY_KEYWORDS
     }
+
+    @ConfigGroup
+    public static class HibernateOrmConfigPersistenceValidator {
+
+        /**
+         * This flag enable/disable hibernate validation.
```
Comment (gastaldi) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java:511]:
```suggestion
         * Enables the Bean Validation integration
```

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/validation/JPAValidationDisabledTestCase.java (hunk 1 above)
Comment (mfpc) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/validation/JPAValidationDisabledTestCase.java]: removed

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java (hunk 2 above)
Comment (mfpc) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java]: done

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/validation/JPAValidationDisabledTestCase.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(MyEntity.class, JPATestValidationResource.class)
+                    .addAsResource("application-validation-disabled.properties", "application.properties"));
+
+    //define an empty import.sql file
+
+    @Test
+    @Transactional
+    public void testValidEntity() {
+        String entityName = MyEntity.ENTITY_NAME_TOO_LONG;
```
Comment (yrodiere) [inline extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/validation/JPAValidationDisabledTestCase.java:27]: This constant is not a value that's too long to fit the entity name, but the message displayed when an entity name is too long.

To actually test that validation is disabled, see how the validation failure is triggered other tests:

```suggestion
        String entityName = "Post method should not persist an entity having a Size constraint of 50 on the name column.";
```

**Review by yrodiere — CHANGES_REQUESTED** [review]

**Review by gsmet — APPROVED** [review]
I made a few minor changes and amended your commit. This should be good to go now.

**Review by yrodiere — APPROVED** [review]

**Comment by yrodiere** [comment]
Looks good, let's merge once CI passes.

Thanks!

**Pull request merged**